featuremanagement/_featuremanager.py

```diff
@@ -230,6 +247,9 @@ def is_enabled(self, feature_flag_id, *args, **kwargs):
         targeting_context = self._build_targeting_context(args)
 
         result = self._check_feature(feature_flag_id, targeting_context, **kwargs)
+        if self._on_feature_evaluated and result.feature.telemetry.enabled:
```

Comment:
Maybe consider to change this condition to the following only, and leave the `feature.telemetry.enabled` check into the impl of `push_telemetry`?
```
if self._on_feature_evaluated
```

Reason:
- `on_feature_evaluated` is a callback function when ff is evaluated. It has nothing to do with what's inside a feature flag, and it doesn't have to be publishing telemetry. So there is no reason we check the evaluation result to determine whether to execute the callback or not.
- `publish_telemetry` is dedicated for telemetry. And it's reasonable that if feature.telemetry.enabled is false we don't send telemetry.

Reply:
line 250 is checking if `_on_feature_evaluated` is not `None` this isn't actually calling the function.

Reply:
I didn't mean that. I mean line 250 should not check `result.feature.telemetry.enabled` because the following block is to call `_on_feature_evaluated`, which doesn't have to be something related to "telemetry". (line 252)

Reply:
`_on_feature_evaluated` is the callback customers setup for the purpose of telemetry. So the idea from the design was that `_on_feature_evaluated` should not be called on flags that don't have telemetry enabled. 

But I do agree that the naming sounds like this shouldn't be limited to telemetry. I think we should either:
1. Rename the callback to something along the lines of "telemetry_callback" 
2. Always call the callback- and let that function handle the `.telemetry.enabled` check.

I prefer option 2, and we add the telemetry check at the start of [`def publish_telemetry`](https://github.com/microsoft/FeatureManagement-Python/pull/24/files/7297743600d12819618130f58fa05fc87d962049#diff-e599fecd01f487185b6b19104e39d7a20d310657896ddeee742af944b129945aR43)

Reply:
@rossgrambo how does the DotNet Feature Management library work for this?

Reply:
When dotnet had the callback- we did not call the callback, but it was specifically a "TelemetryPublisher" that we were not calling. 

Today, if telemetry is disabled- dotnet won't create an activity or event- so there is no non-telemetry callback behavior. But- the reason why is because we didn't want to add too much to their telemetry, which would happen if OpenTelemetry ingested it.

So either:
1. To be in pairity with dotnet- don't call the callback at all
2. Commit to the idea that this callback doesn't have to be telemetry related- and do the check only on the telemetry specific things.

Reply:
@rossgrambo, I think we should default to parity and if DotNet changes we can too.

Reply:
> @rossgrambo, I think we should default to parity and if DotNet changes we can too.

Then I think we should rename the callback to be "telemetry_callback" or something to make it clear that `telemetry.enabled = false` means it won't be called.

Personally- I prefer breaking from .net in regards to this callback- and placing the check inside of `publish_telemetry`.